simpledog_joint_publisher:
  type: "joint_state_controller/JointStateController"
  publish_rate: 200

state_estimate_update:
  type: "balance_controller/StateEstimateController"
  joints: ['front_left_1_joint', 'front_left_2_joint', 'front_left_3_joint', 'front_right_1_joint', 'front_right_2_joint', 'front_right_3_joint', 'rear_right_1_joint', 'rear_right_2_joint', 'rear_right_3_joint', 'rear_left_1_joint', 'rear_left_2_joint', 'rear_left_3_joint']
  real_robot: true
  use_gazebo_feedback: false

single_leg_controller:
  type: "balance_controller/SingleLegController"
  joints: ['front_left_1_joint', 'front_left_2_joint', 'front_left_3_joint', 'front_right_1_joint', 'front_right_2_joint', 'front_right_3_joint', 'rear_right_1_joint', 'rear_right_2_joint', 'rear_right_3_joint', 'rear_left_1_joint', 'rear_left_2_joint', 'rear_left_3_joint']
  legs: ['LF_LEG', 'RF_LEG', 'RH_LEG', 'LH_LEG']
  control_methods: ['end_position','end_position','end_position','end_position']
  real_time: true

base_balance_controller:
  type: "balance_controller/RosBalanceController"
  real_robot: true
  ignore_contact_sensor: true
  joints: ['front_left_1_joint', 'front_left_2_joint', 'front_left_3_joint', 'front_right_1_joint', 'front_right_2_joint', 'front_right_3_joint', 'rear_right_1_joint', 'rear_right_2_joint', 'rear_right_3_joint', 'rear_left_1_joint', 'rear_left_2_joint', 'rear_left_3_joint']
  front_left_1_joint/pid: {p: 300.0, i: 0.01, d: 3.0}
  front_left_2_joint/pid: {p: 300.0, i: 0.01, d: 3.0}
  front_left_3_joint/pid: {p: 300.0, i: 0.01, d: 3.0}
  front_right_1_joint/pid: {p: 300.0, i: 0.01, d: 3.0}
  front_right_2_joint/pid: {p: 300.0, i: 0.01, d: 3.0}
  front_right_3_joint/pid: {p: 300.0, i: 0.01, d: 3.0}
  rear_left_1_joint/pid: {p: 300.0, i: 0.01, d: 3.0}
  rear_left_2_joint/pid: {p: 300.0, i: 0.01, d: 3.0}
  rear_left_3_joint/pid: {p: 300.0, i: 0.01, d: 3.0}
  rear_right_1_joint/pid: {p: 300.0, i: 0.01, d: 3.0}
  rear_right_2_joint/pid: {p: 300.0, i: 0.01, d: 3.0}
  rear_right_3_joint/pid: {p: 300.0, i: 0.01, d: 3.0}

all_joints_position_effort_group_controller:
  type: effort_controllers/JointGroupPositionController
  joints: ['front_left_1_joint', 'front_left_2_joint', 'front_left_3_joint', 'front_right_1_joint', 'front_right_2_joint', 'front_right_3_joint', 'rear_right_1_joint', 'rear_right_2_joint', 'rear_right_3_joint', 'rear_left_1_joint', 'rear_left_2_joint', 'rear_left_3_joint']
  front_left_1_joint/pid: {p: 1000.0, i: 0.01, d: 10.0}
  front_left_2_joint/pid: {p: 1000.0, i: 0.01, d: 10.0}
  front_left_3_joint/pid: {p: 1000.0, i: 0.01, d: 10.0}
  front_right_1_joint/pid: {p: 1000.0, i: 0.01, d: 10.0}
  front_right_2_joint/pid: {p: 1000.0, i: 0.01, d: 10.0}
  front_right_3_joint/pid: {p: 1000.0, i: 0.01, d: 10.0}
  rear_left_1_joint/pid: {p: 1000.0, i: 0.01, d: 10.0}
  rear_left_2_joint/pid: {p: 1000.0, i: 0.01, d: 10.0}
  rear_left_3_joint/pid: {p: 1000.0, i: 0.01, d: 10.0}
  rear_right_1_joint/pid: {p: 1000.0, i: 0.01, d: 10.0}
  rear_right_2_joint/pid: {p: 1000.0, i: 0.01, d: 10.0}
  rear_right_3_joint/pid: {p: 1000.0, i: 0.01, d: 10.0}

all_joints_position_group_controller:
  type: position_controllers/JointGroupPositionController
  joints: ['front_left_1_joint', 'front_left_2_joint', 'front_left_3_joint', 'front_right_1_joint', 'front_right_2_joint', 'front_right_3_joint', 'rear_right_1_joint', 'rear_right_2_joint', 'rear_right_3_joint', 'rear_left_1_joint', 'rear_left_2_joint', 'rear_left_3_joint']
  front_left_1_joint/pid: {p: 1000.0, i: 0.01, d: 10.0}
  front_left_2_joint/pid: {p: 1000.0, i: 0.01, d: 10.0}
  front_left_3_joint/pid: {p: 1000.0, i: 0.01, d: 10.0}
  front_right_1_joint/pid: {p: 1000.0, i: 0.01, d: 10.0}
  front_right_2_joint/pid: {p: 1000.0, i: 0.01, d: 10.0}
  front_right_3_joint/pid: {p: 1000.0, i: 0.01, d: 10.0}
  rear_left_1_joint/pid: {p: 1000.0, i: 0.01, d: 10.0}
  rear_left_2_joint/pid: {p: 1000.0, i: 0.01, d: 10.0}
  rear_left_3_joint/pid: {p: 1000.0, i: 0.01, d: 10.0}
  rear_right_1_joint/pid: {p: 1000.0, i: 0.01, d: 10.0}
  rear_right_2_joint/pid: {p: 1000.0, i: 0.01, d: 10.0}
  rear_right_3_joint/pid: {p: 1000.0, i: 0.01, d: 10.0}

all_joints_velocity_group_controller:
  type: velocity_controllers/JointGroupVelocityController
  joints: ['front_left_1_joint', 'front_left_2_joint', 'front_left_3_joint', 'front_right_1_joint', 'front_right_2_joint', 'front_right_3_joint', 'rear_right_1_joint', 'rear_right_2_joint', 'rear_right_3_joint', 'rear_left_1_joint', 'rear_left_2_joint', 'rear_left_3_joint']
  front_left_1_joint/pid: {p: 1000.0, i: 0.01, d: 10.0}
  front_left_2_joint/pid: {p: 1000.0, i: 0.01, d: 10.0}
  front_left_3_joint/pid: {p: 1000.0, i: 0.01, d: 10.0}
  front_right_1_joint/pid: {p: 1000.0, i: 0.01, d: 10.0}
  front_right_2_joint/pid: {p: 1000.0, i: 0.01, d: 10.0}
  front_right_3_joint/pid: {p: 1000.0, i: 0.01, d: 10.0}
  rear_left_1_joint/pid: {p: 1000.0, i: 0.01, d: 10.0}
  rear_left_2_joint/pid: {p: 1000.0, i: 0.01, d: 10.0}
  rear_left_3_joint/pid: {p: 1000.0, i: 0.01, d: 10.0}
  rear_right_1_joint/pid: {p: 1000.0, i: 0.01, d: 10.0}
  rear_right_2_joint/pid: {p: 1000.0, i: 0.01, d: 10.0}
  rear_right_3_joint/pid: {p: 1000.0, i: 0.01, d: 10.0}

all_joints_effort_group_controller:
  type: effort_controllers/JointGroupEffortController
  joints: ['front_left_1_joint', 'front_left_2_joint', 'front_left_3_joint', 'front_right_1_joint', 'front_right_2_joint', 'front_right_3_joint', 'rear_right_1_joint', 'rear_right_2_joint', 'rear_right_3_joint', 'rear_left_1_joint', 'rear_left_2_joint', 'rear_left_3_joint']
  front_left_1_joint/pid: {p: 1000.0, i: 0.01, d: 10.0}
  front_left_2_joint/pid: {p: 1000.0, i: 0.01, d: 10.0}
  front_left_3_joint/pid: {p: 1000.0, i: 0.01, d: 10.0}
  front_right_1_joint/pid: {p: 1000.0, i: 0.01, d: 10.0}
  front_right_2_joint/pid: {p: 1000.0, i: 0.01, d: 10.0}
  front_right_3_joint/pid: {p: 1000.0, i: 0.01, d: 10.0}
  rear_left_1_joint/pid: {p: 1000.0, i: 0.01, d: 10.0}
  rear_left_2_joint/pid: {p: 1000.0, i: 0.01, d: 10.0}
  rear_left_3_joint/pid: {p: 1000.0, i: 0.01, d: 10.0}
  rear_right_1_joint/pid: {p: 1000.0, i: 0.01, d: 10.0}
  rear_right_2_joint/pid: {p: 1000.0, i: 0.01, d: 10.0}
  rear_right_3_joint/pid: {p: 1000.0, i: 0.01, d: 10.0}

front_left_1_joint_position_controller:
  type: effort_controllers/JointPositionController
  joint: front_left_1_joint
  pid: {p: 1000.0, i: 0.01, d: 10.0}

front_left_2_joint_position_controller:
  type: effort_controllers/JointPositionController
  joint: front_left_2_joint
  pid: {p: 1000.0, i: 0.01, d: 10.0}

front_left_3_joint_position_controller:
  type: effort_controllers/JointPositionController
  joint: front_left_3_joint
  pid: {p: 1000.0, i: 0.01, d: 10.0}

front_right_1_joint_position_controller:
  type: effort_controllers/JointPositionController
  joint: front_right_1_joint
  pid: {p: 1000.0, i: 0.01, d: 10.0}

front_right_2_joint_position_controller:
  type: effort_controllers/JointPositionController
  joint: front_right_2_joint
  pid: {p: 1000.0, i: 0.01, d: 10.0}

front_right_3_joint_position_controller:
  type: effort_controllers/JointPositionController
  joint: front_right_3_joint
  pid: {p: 1000.0, i: 0.01, d: 10.0}

rear_right_1_joint_position_controller:
  type: effort_controllers/JointPositionController
  joint: rear_right_1_joint
  pid: {p: 1000.0, i: 0.01, d: 10.0}

rear_right_2_joint_position_controller:
  type: effort_controllers/JointPositionController
  joint: rear_right_2_joint
  pid: {p: 1000.0, i: 0.01, d: 10.0}

rear_right_3_joint_position_controller:
  type: effort_controllers/JointPositionController
  joint: rear_right_3_joint
  pid: {p: 1000.0, i: 0.01, d: 10.0}

rear_left_1_joint_position_controller:
  type: effort_controllers/JointPositionController
  joint: rear_left_1_joint
  pid: {p: 1000.0, i: 0.01, d: 10.0}

rear_left_2_joint_position_controller:
  type: effort_controllers/JointPositionController
  joint: rear_left_2_joint
  pid: {p: 1000.0, i: 0.01, d: 10.0}

rear_left_3_joint_position_controller:
  type: effort_controllers/JointPositionController
  joint: rear_left_3_joint
  pid: {p: 1000.0, i: 0.01, d: 10.0}
